datafusion/core/tests/dataframe/mod.rs

```diff
@@ -305,6 +305,17 @@ async fn select_columns() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn select_columns_with_nonexistent_columns() -> Result<()> {
+    let t = test_table().await?;
+    let t2 = t.select_columns(&["canada", "c2", "rocks"]);
+    let Err(DataFusionError::SchemaError(_, _)) = t2 else {
+        return internal_err!("select_columns with nonexistent columns should error");
+    };
```

Comment:
We could verify the actual error message here (or using `assert_snapshot!`):

```suggestion
    assert_contains!(
        t2.unwrap_err().strip_backtrace(),
        "Schema error: No field named canada"
    );
```